src/QirRuntime/test/unittests/QirRuntimeTests.cpp

```diff
@@ -841,4 +865,64 @@ TEST_CASE("Unpacking input tuples of nested callables (case1)", "[qir_support]")
     quantum__rt__qubit_release_array(controlsInner);
     quantum__rt__array_unreference(controlsInner);
     quantum__rt__qubit_release(target);
+}
+
+TEST_CASE("Allocation tracking for arrays", "[qir_support]")
+{
+    InitializeQirContext(nullptr /*don't need a simulator*/, true /*track allocations*/);
+
+    QirArray* bounce = quantum__rt__array_create_1d(1, 1);
+    quantum__rt__array_unreference(bounce);
+    CHECK_THROWS(quantum__rt__array_reference(bounce));
+
+    QirArray* releaseTwice = quantum__rt__array_create_1d(1, 1);
+    quantum__rt__array_unreference(releaseTwice);
+    CHECK_THROWS(quantum__rt__array_unreference(releaseTwice));
+
+    QirArray* maybeLeaked = quantum__rt__array_create_1d(1, 1);
+    CHECK_THROWS(ReleaseQirContext());
+
+    quantum__rt__array_unreference(maybeLeaked);
+    ReleaseQirContext();

```

Comment:
`CHECK_NO_THROW(ReleaseQirContext())`?

Reply:
The check_nothrow could, in theory, allow other tests to run rather than terminate on exception, however, if an exception is thrown, the context won't be cleaned up and other tests would run into asserts about that... I suppose, it's still good to have check_nothrow for documentation purposes here.

Reply:
If terminating upon unexpected exception provides a better diagnostic then it probably makes sense to leave as is. I saw such situations earlier and agree, check_nothrow may be a worse idea.